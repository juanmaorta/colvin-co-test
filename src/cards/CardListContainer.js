import React, { Component } from 'react'
import { connect } from 'react-redux'
import PropTypes from 'prop-types'
import Button from '@material-ui/core/Button'
import Typography from '@material-ui/core/Typography'
import { withStyles } from '@material-ui/core/styles'

import { generateRandomEmptyCard, getBlackCards } from '../util/cards'
import CardList from './components/CardList'
import {
  addCard,
  deleteCard,
  editCard
} from './actions'
import styles from '../styles'

const mapStateToProps = (state) => ({
  cards: getBlackCards(state.cards)
})

const mapDispatchToProps = (dispatch) => ({
  deleteCard: (cardId) => {
    return dispatch(deleteCard(cardId))
  },
  editCard: (card) => {
    return dispatch(editCard(card))
  },
  addCard: (card) => {
    return dispatch(addCard(generateRandomEmptyCard()))
  }
})

export class CardListContainer extends Component {
  render () {
    const {
      addCard,
      cards,
      classes,
      deleteCard,
      editCard
    } = this.props

    return (
      <div className={classes.heroUnit}>
        <Typography variant='h4' align='center' color='textPrimary' gutterBottom>
          Choose your black card
        </Typography>
        <CardList
          cards={cards}
          deleteCard={deleteCard}
          editCard={editCard}
        />
        <div className={classes.buttons}>
          <Button variant='contained' onClick={addCard}>Add a black card</Button>
        </div>
      </div>
    )
  }
}

CardListContainer.propTypes = {
  classes: PropTypes.object.isRequired
}

export default withStyles(styles)(
  connect(mapStateToProps, mapDispatchToProps)(CardListContainer)
)
