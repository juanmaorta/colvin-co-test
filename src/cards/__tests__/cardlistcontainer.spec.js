/* global expect, describe, it, xit */
import React from 'react'
import { shallow } from 'enzyme'

import { CardListContainer } from '../CardListContainer'
import * as stubs from './stubs'

describe('CardListContainer snapshots', () => {
  it('should match for CardListContainer component', () => {
    const wrapper = shallow(
      <CardListContainer
        classes={stubs.classes}
        cards={stubs.cards}
      />
    )
    expect(wrapper).toMatchSnapshot()
  })
})
