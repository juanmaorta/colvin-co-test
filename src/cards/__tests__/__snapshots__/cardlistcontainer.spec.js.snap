// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardListContainer snapshots should match for CardListContainer component 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CardListContainer
    cards={
        Array [
            Object {
              "id": "3c6f4977-c58b-460f-98ad-acd2fae3a2ce",
              "text": "GoldFinger: He's the man, the man with...",
            },
            Object {
              "id": "da1f1278-3f91-4529-8e55-81b4ffd621ac",
              "text": "Instead of coal, Santa now gives the bad children...",
            },
            Object {
              "id": "d50a2288-c93f-448f-8af6-3dc1b7268606",
              "text": "Knock knock. Who's there?",
            },
            Object {
              "id": "2198fcf9-cb27-48e4-befa-aa77da6f1145",
              "parentId": "3c6f4977-c58b-460f-98ad-acd2fae3a2ce",
              "text": "GoldFinger kid",
            },
          ]
    }
    classes={Object {}}
    deleteCard={[Function]}
    editCard={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <WithStyles(Typography)
          align="center"
          color="textPrimary"
          gutterBottom={true}
          variant="h4"
>
          Choose your black card
</WithStyles(Typography)>,
        <CardList
          cards={
                    Array [
                              Object {
                                "id": "3c6f4977-c58b-460f-98ad-acd2fae3a2ce",
                                "text": "GoldFinger: He's the man, the man with...",
                              },
                              Object {
                                "id": "da1f1278-3f91-4529-8e55-81b4ffd621ac",
                                "text": "Instead of coal, Santa now gives the bad children...",
                              },
                              Object {
                                "id": "d50a2288-c93f-448f-8af6-3dc1b7268606",
                                "text": "Knock knock. Who's there?",
                              },
                              Object {
                                "id": "2198fcf9-cb27-48e4-befa-aa77da6f1145",
                                "parentId": "3c6f4977-c58b-460f-98ad-acd2fae3a2ce",
                                "text": "GoldFinger kid",
                              },
                            ]
          }
          deleteCard={[Function]}
          editCard={[Function]}
/>,
        <div
          className={undefined}
>
          <WithStyles(Button)
                    onClick={undefined}
                    variant="contained"
          >
                    Add a black card
          </WithStyles(Button)>
</div>,
      ],
      "className": undefined,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "align": "center",
          "children": "Choose your black card",
          "color": "textPrimary",
          "gutterBottom": true,
          "variant": "h4",
        },
        "ref": null,
        "rendered": "Choose your black card",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "cards": Array [
            Object {
              "id": "3c6f4977-c58b-460f-98ad-acd2fae3a2ce",
              "text": "GoldFinger: He's the man, the man with...",
            },
            Object {
              "id": "da1f1278-3f91-4529-8e55-81b4ffd621ac",
              "text": "Instead of coal, Santa now gives the bad children...",
            },
            Object {
              "id": "d50a2288-c93f-448f-8af6-3dc1b7268606",
              "text": "Knock knock. Who's there?",
            },
            Object {
              "id": "2198fcf9-cb27-48e4-befa-aa77da6f1145",
              "parentId": "3c6f4977-c58b-460f-98ad-acd2fae3a2ce",
              "text": "GoldFinger kid",
            },
          ],
          "deleteCard": [Function],
          "editCard": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <WithStyles(Button)
            onClick={undefined}
            variant="contained"
>
            Add a black card
</WithStyles(Button)>,
          "className": undefined,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "Add a black card",
            "onClick": undefined,
            "variant": "contained",
          },
          "ref": null,
          "rendered": "Add a black card",
          "type": [Function],
        },
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <WithStyles(Typography)
            align="center"
            color="textPrimary"
            gutterBottom={true}
            variant="h4"
>
            Choose your black card
</WithStyles(Typography)>,
          <CardList
            cards={
                        Array [
                                    Object {
                                      "id": "3c6f4977-c58b-460f-98ad-acd2fae3a2ce",
                                      "text": "GoldFinger: He's the man, the man with...",
                                    },
                                    Object {
                                      "id": "da1f1278-3f91-4529-8e55-81b4ffd621ac",
                                      "text": "Instead of coal, Santa now gives the bad children...",
                                    },
                                    Object {
                                      "id": "d50a2288-c93f-448f-8af6-3dc1b7268606",
                                      "text": "Knock knock. Who's there?",
                                    },
                                    Object {
                                      "id": "2198fcf9-cb27-48e4-befa-aa77da6f1145",
                                      "parentId": "3c6f4977-c58b-460f-98ad-acd2fae3a2ce",
                                      "text": "GoldFinger kid",
                                    },
                                  ]
            }
            deleteCard={[Function]}
            editCard={[Function]}
/>,
          <div
            className={undefined}
>
            <WithStyles(Button)
                        onClick={undefined}
                        variant="contained"
            >
                        Add a black card
            </WithStyles(Button)>
</div>,
        ],
        "className": undefined,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "align": "center",
            "children": "Choose your black card",
            "color": "textPrimary",
            "gutterBottom": true,
            "variant": "h4",
          },
          "ref": null,
          "rendered": "Choose your black card",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "cards": Array [
              Object {
                "id": "3c6f4977-c58b-460f-98ad-acd2fae3a2ce",
                "text": "GoldFinger: He's the man, the man with...",
              },
              Object {
                "id": "da1f1278-3f91-4529-8e55-81b4ffd621ac",
                "text": "Instead of coal, Santa now gives the bad children...",
              },
              Object {
                "id": "d50a2288-c93f-448f-8af6-3dc1b7268606",
                "text": "Knock knock. Who's there?",
              },
              Object {
                "id": "2198fcf9-cb27-48e4-befa-aa77da6f1145",
                "parentId": "3c6f4977-c58b-460f-98ad-acd2fae3a2ce",
                "text": "GoldFinger kid",
              },
            ],
            "deleteCard": [Function],
            "editCard": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <WithStyles(Button)
              onClick={undefined}
              variant="contained"
>
              Add a black card
</WithStyles(Button)>,
            "className": undefined,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Add a black card",
              "onClick": undefined,
              "variant": "contained",
            },
            "ref": null,
            "rendered": "Add a black card",
            "type": [Function],
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
